lib/src/Base/Stat/openturns/CorrelationAnalysis.hxx

```diff
@@ -34,37 +35,69 @@ BEGIN_NAMESPACE_OPENTURNS
  *
  */
 
-namespace CorrelationAnalysis
+class OT_API CorrelationAnalysis
+  : public PersistentObject
 {
-/** Compute the Pearson correlation coefficient between the input sample and the output sample */
-OT_API Point PearsonCorrelation(const Sample & inputSample,
-                                const Sample & outputSample);
-
-/** Compute the Spearman correlation coefficient between the input sample and the output sample */
-OT_API Point SpearmanCorrelation(const Sample & inputSample,
-                                 const Sample & outputSample);
-
-/** Compute the Standard Regression Coefficients (SRC) between the input sample and the output sample */
-OT_API Point SRC(const Sample & inputSample,
-                 const Sample & outputSample,
-                 const Bool normalize = false);
-
-/** Compute the signed Standard Regression Coefficients (SRC) between the input sample and the output sample */
-OT_API Point SignedSRC(const Sample & inputSample,
-                       const Sample & outputSample);
-
-/** Compute the Partial Correlation Coefficients (PCC) between the input sample and the output sample */
-OT_API Point PCC(const Sample & inputSample,
-                 const Sample & outputSample);
-
-/** Compute the Standard Rank Regression Coefficients (SRRC) between the input sample and the output sample */
-OT_API Point SRRC(const Sample & inputSample,
-                  const Sample & outputSample,
-                  const Bool normalize = false);
-
-/** Compute the Partial Rank Correlation Coefficients (PRCC) between the input sample and the output sample */
-OT_API Point PRCC(const Sample & inputSample,
-                  const Sample & outputSample);
+  CLASSNAME
+public:
+
+  /** Default constructor */
+  CorrelationAnalysis();
+
+  /** Standard constructor */
+  CorrelationAnalysis(const Sample & inputSample,
+                      const Sample & outputSample);
+
+  /** Virtual constructor */
+  CorrelationAnalysis * clone() const override;
+
+  /** String converter */
+  String __repr__() const override;
+
+  /** Compute the Pearson correlation coefficient between the input sample and the output sample */
+  const Point computePearsonCorrelation();
```

Comment:
avoid const for return types when its not a pointer

Reply:
Oh yes, that's what I was trying to do.

Reply:
Fixed.